### PR TITLE
BCDA-4081: Migrate BCDA SSAS App to Go Modules

### PR DESCRIPTION
**Pull request by adamayd**
### Fixes [BCDA-4081](https://jira.cms.gov/browse/BCDA-4081)

Since version 1.11 Go has switched to Modules for dependencies and `dep` (our current dependency manager) has been deprecated by the authors.  

### Proposed Changes

* Remove `dep` dependancy manager
* Migrate current dependencies to Go Modules of the the same version

### Change Details

Removing the `dep` installation and `dep ensure` commands from the Dockerfiles.  Running `go mod init` to initialize the project.  Running `go mod tidy` to populate the `$GOPATH/src/pkg/mod` directory with the vendor files.  Check versions between the ~~`Gopkg.toml`~~ `Gopkg.lock` and `go.mod` file to ensure version parity from the conversion.  Remove the 'vendor/` directory and Gopkg.* files.

### Security Implications

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [x] no PHI/PII is affected by this change

### Acceptance Validation

- [x] `dep` is no longer installed on build
- [x] `go.mod` has the correct dependencies and versions prior to initialization
- [x] Documentation has been updated.
- [x] The project builds and tests locally without issue
- [x] The project builds and deploys to `dev` successfully
Screenshot from 2021-01-15 09-45-43
- [x] Smoke tests pass in `dev` after deployment
Screenshot from 2021-01-15 09-45-33

### Feedback Requested

Best Practices, Better Ideas.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module github.com/CMSgov/bcda-ssas-app
+
+go 1.15
+
+require (
```
Comment (mtrang1263) [inline go.mod:5]: Want to doublecheck, these deps match up with what used to exist in the Gopkg.lock file, right?

**Review by mtrang1263 — COMMENTED** [review]
From this PR: https://github.com/CMSgov/bcda-app/pull/615, looks like we're going to update the README.

Do you think it's worthwhile to include a snippet in this README.md to describes steps on adding, upgrading dependencies? Some examples might be helpful.

File: go.mod (hunk 1 above)
Comment (adamayd) [reply go.mod:5]: Okay, I will check again.

Comment (adamayd) [reply go.mod:5]: Offline conversation decided to manage dependencies based on the last used commit per `dep` and update each package separately with necessary code changes.  This method will assure we are using the same exact package to the commit during this change over to Go Modules.  Will update and submit changes before resolving conversation.

File: README.md [hunk 2]
```diff
@@ -74,6 +74,10 @@ To avoid committing and pushing unencrypted secret files, use the included `scri
 cp ops/pre-commit .git/hooks
 ```
 
+## Go Modules
```
Comment (katyasoup) [inline README.md:77]: 😄

**Review by katyasoup — APPROVED** [review]
LGTM!

**Review by mtrang1263 — APPROVED** [review]
Nice job!

**Review by SMLuthi — APPROVED** [review]
LGTM!

**Pull request merged**